lib/secretsManager.js

```diff
@@ -1,77 +1,154 @@
-let DUMMY_UUID_GEN = 0
-let DUMMY_STORE = []
-
-/**
- * Find all secrets, applying a selector if provided.
- * When provided, all where clause must be fulfilled (logical AND).
- * By default, returns all secrets
- * @async
- * @param {Object} where - hash of where clauses returned secrets must fulfill with
- * @returns {Array<Object>} list (may be empty) of matching secrets
- */
-export async function find(where = {}) {
-  let secrets = []
-
-  try {
-    const props = Object.keys(where)
-    secrets = DUMMY_STORE.filter(secret =>
-      props.every(prop => secret[prop] === where[prop])
+import uuid from 'uuid-random'
+import aesjs from 'aes-js'
+import * as SecureStore from 'expo-secure-store'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+const encryptionPassphraseStorageKey = 'encryptionPassphrase'
+const secretStorageKey = 'secrets'
+
+export default class SecretsManager {
+  constructor() {
+    this.encryptionPassphrase = null
+  }
+
+  async init() {
+    let getEncryptionKey = AsyncStorage.getItem
+    let setEncryptionKey = AsyncStorage.setItem
+
+    if (SecureStore.isAvailableAsync()) {
+      getEncryptionKey = SecureStore.getItemAsync
+      setEncryptionKey = SecureStore.setItemAsync
+    }
+
+    const encryptionPassphrase = await getEncryptionKey(
+      encryptionPassphraseStorageKey
     )
-  } catch (err) {
-    console.error('Failed to read secrets from data store', err)
+
+    if (encryptionPassphrase) {
+      this.encryptionPassphrase = encryptionPassphrase
+      return
+    }
+
+    this.encryptionPassphrase = this.generateEncryptionPassphrase()
+
+    await setEncryptionKey(
+      encryptionPassphraseStorageKey,
+      this.encryptionPassphrase
+    )
+
+    await this.set([])
   }
 
-  return secrets
-}
+  generateEncryptionPassphrase() {
+    return JSON.stringify(
+      Array.from({ length: 32 }).map(() => Math.floor(Math.random() * 256))
+    )
+  }
+
+  async encrypt(secrets) {
+    const secretsBytes = aesjs.utils.utf8.toBytes(JSON.stringify(secrets))
+    const aesCtr = new aesjs.ModeOfOperation.ctr(
+      Uint8Array.from(JSON.parse(this.encryptionPassphrase)),
+      new aesjs.Counter(5)
```

Comment:
does this create always the same thing? if so, let's extract it in a module scoped constant and give it a meaningful name

Reply:
It seems to be something we really need to create and recreate every time.